### DISCUSSION TITLE
getFirstListItem() returns error when record doesnt exists

### DISCUSSION THREAD

**Discussion by Marius555** [pocketbase/pocketbase#3648] [Q&A]
hi, im trying to check if the record already exists before submiting, i would assume that getFirstListItem() will return false or something when record is not found, but i get an error, "The requested resource wasn't found." is this how it suppose to work?

```js
const SimpleProfiler = async (data) => {
    const pb = new PocketBase('http://127.0.0.1:8090');
    try {
        const cockies = cookies().get("pb_auth")
        pb.authStore.loadFromCookie(`${cockies.name}=${cockies.value}`)
        const mod_data = await { ...data, UserId: pb.authStore.model.id }
        const req_user = await pb.collection("SimpleProfiler").getFirstListItem(`UserId="${await mod_data.UserId}"`);
        if (!req_user) {
            await pb.collection('SimpleProfiler').create(await mod_data)
        }
        else {
            console.log("record already exists")
        }

    } catch (error) {
        return console.log("semething went wrong", error)
    }
}
```

**Comment by ganigeorgiev** [comment]
Yes, it behaves similar to `getOne` (_see also the [jsdoc of the method](https://github.com/pocketbase/js-sdk/blob/98b5fa014a27b88a6f25d39d741b10f821d0a24a/src/services/utils/CrudService.ts#L77-L87)_).

**Reply by ganigeorgiev** [reply]
btw. if your use case require to have "SimpleProfiler" records with unique "UserId" field, then you don't have to send a request to perform the check manually and instead can create a unique index for your "UserId" field (_it can be created from the _"Unique constraints and indexes"_ section under the schema fields in the Admin UI_).

**Reply by Marius555** [reply]
okey, then what is the correct way to check if user exists? like i cant use if because its returns an error, so i have to use try block inside try block? like how is this works? i come from python background, in there if record doesn't exists its just returns None so you can use it in if block...

**Reply by ganigeorgiev** [reply]
See the above comment. Also please check [JS SDK Error handling](https://github.com/pocketbase/js-sdk#error-handling).

**Reply by Marius555** [reply]
okey, i found section "Unique constraints and indexes" and how doest it work? like there is documentation for how to use it?

**Reply by nedpals** [reply]
> okey, i found section "Unique constraints and indexes" and how doest it work? like there is documentation for how to use it?

PB's design to this is similar to how a relational database works. Under the hood, unique constraints just marks the column as "unique" so as it does not have any duplicate entries upon creating. The side effect of making a column "unique" is faster lookups of the records (if i am right)
